src/serializer/ResidualHeapSerializer.js

```diff
@@ -108,11 +108,11 @@ export class ResidualHeapSerializer {
     this._descriptors = new Map();
     this.needsEmptyVar = false;
     this.needsAuxiliaryConstructor = false;
-    this.valueNameGenerator = this.preludeGenerator.createNameGenerator("_");
-    this.descriptorNameGenerator = this.preludeGenerator.createNameGenerator("$$");
-    this.factoryNameGenerator = this.preludeGenerator.createNameGenerator("$_");
-    this.intrinsicNameGenerator = this.preludeGenerator.createNameGenerator("$i_");
-    this.functionNameGenerator = this.preludeGenerator.createNameGenerator("$f_");
+    this.valueNameGenerator = this.preludeGenerator.getValueNameGenerator();
```

Comment:
With --inlineExpressions, we create two instances of `ResidualHeapSerializer`. Before your change, they would create separate name generators, each time starting from 0. After your change, they would the name generators would live on, and the second time around we get larger numbers. Besides resulting in unnecessarily long identifiers, this might also potentially cause some minor other differences in the code generation which would cause the identifier-counting scheme to diverge. So in a nutshell, I don't think you should do this change.

Reply:
You do something very special with the `valueNameGenerator`. Still, my comment applies to all other generators.

Reply:
I hate this inlineExpressions design that create name generators and lots of code and completely throw them away...
Anyway, under this situation, I would still make valueNameGenerator singleton while reverts all other name generators to be local.
For debug names, they should get name from ResidualHeapValueIdentifiers class which will be persisted across two passes.